File: model/empresa.py
class Empresa:
    def __init__(self, nombre):
        self.nombre = nombre
        self.departamentos = []

    def agregar_departamento(self, d):
        self.departamentos.append(d)

    def buscar_departamento(self, nombre):
        for d in self.departamentos:
            if d.nombre == nombre:
                return d
        return None

    def listar_departamentos(self):
        for d in self.departamentos:
            print(d.nombre)

    def mostrar_todo(self):
        for d in self.departamentos:
            print("\n" + d.nombre)
            d.listar_empleados()

File: Empresa.java
import java.util.ArrayList;

public class Empresa {
    private String nombre;
    private ArrayList<Departamento> departamentos;

    public Empresa(String nombre) {
        this.nombre = nombre;
        departamentos = new ArrayList<>();
    }

    public void agregarDepartamento(Departamento d) {
        departamentos.add(d);
    }

    public Departamento buscarDepartamento(String nombre) {
        for (Departamento d : departamentos) {
            if (d.getNombre().equalsIgnoreCase(nombre)) {
                return d;
            }
        }
        return null;
    }

    public void listarDepartamentos() {
        if (departamentos.isEmpty()) {
            System.out.println("No hay departamentos.");
            return;
        }

        for (Departamento d : departamentos) {
            System.out.println("- " + d.getNombre());
        }
    }

    public void mostrarTodo() {
        for (Departamento d : departamentos) {
            System.out.println("\nDepartamento: " + d.getNombre());
            d.listarEmpleados();
        }
    }
}
